doctr/models/recognition/master/pytorch.py

```diff
@@ -109,10 +109,10 @@ def make_source_and_target_mask(
         # [True, True, True, ..., False, False, False] -> False is masked
         target_pad_mask = (target != self.vocab_size + 2).unsqueeze(1).unsqueeze(1)  # (N, 1, 1, max_length)
         target_length = target.size(1)
-        # sub mask filled diagonal with 1 = see 0 = masked (max_length, max_length)
-        target_sub_mask = torch.tril(
-            torch.ones((target_length, target_length), dtype=torch.bool, device=source.device), diagonal=0
-        )
+        # sub mask filled diagonal with True = see and False = masked (max_length, max_length)
+        target_sub_mask = (
+            torch.triu(torch.ones((target_length, target_length), device=source.device)) == 1
```

Comment:
If this is compatible, those options are much cleaner (best being the first):
```python
torch.triu(torch.ones((target_length, target_length), dtype=torch.bool, device=source.device)).transpose(0, 1)
```

```python
torch.triu(torch.ones((target_length, target_length), device=source.device)).to(dtype=torch.bool).transpose(0, 1)
```

Reply:
Unfortunately triu works currently only with float in onnxruntime 1.11.1 opset 14 i have added a comment

Reply:
but your .to example works fine with tril nice :)